src/Generators/Microsoft.Gen.Logging/Parsing/Parser.cs

```diff
@@ -592,35 +592,67 @@ private void CheckMethodParametersAreUnique(LoggingMethod lm, Dictionary<Logging
         return null;
     }
 
-    private (string? field, IFieldSymbol? secondLoggerField, bool isNullable) FindField(SemanticModel sm, TypeDeclarationSyntax classDec, ITypeSymbol symbol)
+    private (string? member, ISymbol? secondLoggerMember, bool isNullable) FindMember(SemanticModel sm, TypeDeclarationSyntax classDec, ITypeSymbol symbol)
     {
-        string? field = null;
+        string? member = null;
         bool isNullable = false;
 
-        foreach (var m in classDec.Members)
+        INamedTypeSymbol? type = sm.GetDeclaredSymbol(classDec);
+        var originType = type!;
+        while (type != null)
```

Comment:
I would also suggest skipping members that are implicitly added by a compiler. This API can be used: https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.isymbol.isimplicitlydeclared?view=roslyn-dotnet-4.7.0#microsoft-codeanalysis-isymbol-isimplicitlydeclared

Reply:
Oh, you're already checking `AssociatedSymbol`. Never mind then...